core/src/Kokkos_Macros.hpp

```diff
@@ -530,6 +530,19 @@
 #endif
 #define KOKKOS_DEPRECATED_TRAILING_ATTRIBUTE
 
+#define KOKKOS_IMPL_STRINGIFY(x) #x
+#define KOKKOS_IMPL_TOSTRING(x) KOKKOS_IMPL_STRINGIFY(x)
+
+#ifdef _MSC_VER
+#define KOKKOS_DO_PRAGMA(x) __pragma(x)
+#define KOKKOS_WARNING(desc) \
+  KOKKOS_DO_PRAGMA(message(  \
+      __FILE__ "(" KOKKOS_IMPL_TOSTRING(__LINE__) ") : warning: " #desc))
+#else
+#define KOKKOS_DO_PRAGMA(x) _Pragma(#x)
+#define KOKKOS_WARNING(desc) KOKKOS_DO_PRAGMA(message(#desc))
```

Comment:
Would you remind me why this is it not just `#warning <msg>`?

Reply:
That's what we are using in deal.II. Just `#warning` wasn't portable enough to also work with `MSVC` so we needed some kind of preprocessor macro and that's why we came up with this partly taken from http://goodliffe.blogspot.com/2009/07/c-how-to-say-warning-to-visual-studio-c.html. `gcc` and `clang` also don't like the `MSVC` version. I'm happy to replace it if there are better suggestions.